src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -2,28 +2,46 @@ import React from 'react';
 import SimpleToast from 'react-native-simple-toast';
 import {launchImageLibrary} from 'react-native-image-picker';
 
-import StringConstant from '../../../utils/string/StringConstant';
+import {Linking} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+import {useRecoilState, atom} from 'recoil';
 import {Context} from '../../../context';
 import {getChatName} from '../../../utils/string/StringUtils';
 import {requestExternalStoragePermission} from '../../../utils/permission';
+import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import {CHANNEL_GROUP, CHAT_ATOM} from '../../../hooks/core/constant';
+import StringConstant from '../../../utils/string/StringConstant';
 import {uploadFile} from '../../../service/file';
 
-const useGroupSetting = ({navigation, route}) => {
-  const [groupChatState] = React.useContext(Context).groupChat;
+const chatAtom = atom({
+  key: CHAT_ATOM,
+  default: {
+    selectedChannel: null
+  }
+});
 
+const useGroupSetting = ({route}) => {
+  const [groupChatState] = React.useContext(Context).groupChat;
+  const navigation = useNavigation();
   const [channelState] = React.useContext(Context).channel;
   const [profile] = React.useContext(Context).profile;
   const [participants, setParticipants] = React.useState([]);
   const {channel} = channelState;
   const [groupName, setGroupName] = React.useState(
     getChatName(route.params.name, profile.myProfile.username) || 'Set Group Name'
   );
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const [loadingUpdate, setLoadingUpdate] = React.useState(false);
   const [countUser, setCountUser] = React.useState(0);
   const [changeName, setChangeName] = React.useState(false);
   const [changeImage, setChangeImage] = React.useState(false);
   const [base64Profile, setBase64Profile] = React.useState('');
   const [urlImage, setUrlImage] = React.useState(channel?.data?.image);
   const [isLoading, setIsLoading] = React.useState(false);
+  const [chat, setChat] = useRecoilState(chatAtom);
+  const {selectedChannel} = chat;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces the use of Recoil for state management, specifically for handling chat state. However, it's important to ensure that this change is consistent across the entire application. If Recoil is only used in this one component while the rest of the app uses Context API, it could lead to confusion and inconsistency. If Recoil is being introduced as a new state management solution, consider refactoring other parts of the app to use Recoil as well.

```diff
- const [chat, setChat] = useRecoilState(chatAtom);
+ const [chat, setChat] = React.useContext(Context).chat;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
for chat we use recoil